actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/service/impl/UserSelfDeclarationServiceImpl.java

```diff
@@ -0,0 +1,204 @@
+package org.sunbird.user.service.impl;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.collections.CollectionUtils;
+import org.sunbird.cassandra.CassandraOperation;
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.response.Response;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.LoggerUtil;
+import org.sunbird.common.request.RequestContext;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.helper.ServiceFactory;
+import org.sunbird.models.user.UserDeclareEntity;
+import org.sunbird.user.dao.UserSelfDeclarationDao;
+import org.sunbird.user.dao.impl.UserSelfDeclarationDaoImpl;
+import org.sunbird.user.service.UserSelfDeclarationService;
+
+public class UserSelfDeclarationServiceImpl implements UserSelfDeclarationService {
+
+  private LoggerUtil logger = new LoggerUtil(UserSelfDeclarationServiceImpl.class);
+  private static UserSelfDeclarationService selfDeclarationService = null;
+  private CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+  private ObjectMapper mapper = new ObjectMapper();
+  private static UserSelfDeclarationDao userSelfDeclarationDao =
+      UserSelfDeclarationDaoImpl.getInstance();
+
+  public static UserSelfDeclarationService getInstance() {
+    if (selfDeclarationService == null) {
+      selfDeclarationService = new UserSelfDeclarationServiceImpl();
+    }
+    return selfDeclarationService;
+  }
+
+  public Response saveUserSelfDeclareAttributes(
+      Map<String, Object> requestMap, RequestContext context) {
+    List<UserDeclareEntity> selfDeclaredFields =
```

Comment:
Please consider about legs.

Reply:
Entry and exit logs has to be in API level, so another ticket will take care of this for all APIs